fms_core/resources.py

```diff
@@ -114,6 +114,21 @@ def import_field(self, field, obj, data, is_m2m=False):
 
         elif field.attribute == 'volume_history':
             obj.volume_history = [create_volume_history("update", data["Volume (uL)"])]
+        # if sample is in tube
+        elif field.attribute == 'container':
+            if data['Container Kind'] == 'tube':
+                tube_container_data = dict(
+                    kind=data['Container Kind'],
+                    name=data['Container Name'],
+                    barcode=data['Container Barcode'],
+                    location=Container.objects.get(barcode=data['Location Barcode']),
+                    coordinates=data['Location Coord']
+                )
+                try:
+                    container = Container.objects.get(**tube_container_data)
```

Comment:
Since you're not adding any extra tube container data here, we can just use get_or_create